### DISCUSSION TITLE
Google Batch: /usr/bin/fusermount: fuse device not found, try 'modprobe fuse' first

### DISCUSSION THREAD

**Discussion by VasLem** [nextflow-io/nextflow#5236] [Q&A] [answered]
## Bug report 
I am trying to mount some Google Cloud directories through a docker entrypoint script I have made (the commented regions was an effort to make it working):
```
#!/bin/bash
set -e
opts=
if gcloud projects list &> /dev/null; then
    echo -e "GCloud credentials not needed to mount."
    opts=
else
    opts="--key-file /tmp/keys/gcloud_credentials.json"
fi
echo -e "Mounting GCP folders.."
for d in bin lib data tests
do
    # if [ -d /mnt/disks/$MAIN_BUCKET ] # within google cloud
    # then
    #     if [ -d /mnt/disks/$MAIN_BUCKET/$d/$1 ]
    #     then
    #         ln -sf /mnt/disks/$MAIN_BUCKET/$d/$1 $d
    #     fi
    # else
        gsutil ls $ops gs://$MAIN_BUCKET/$d/$1 &> /dev/null && {
            mkdir -p $d
            # Allow only read and execute, not write or edit
            chmod 555 ${d}
            gcsfuse --file-mode=555 --dir-mode=555  --implicit-dirs $opts --only-dir $d/$1/  $MAIN_BUCKET $d &> /dev/null ||
                gcsfuse  --foreground --debug_fs --debug_fuse --file-mode=555  --dir-mode=555  --implicit-dirs $opts --only-dir $d/$1/  $MAIN_BUCKET $d # to print error
        }
    # fi
done
echo -e "Mounting complete."
shift
set +e
exec "$@"
```

The local profile solution (not on Google Batch) works fine with this configuration
```
docker{
      // in order to be able to mount with gcsfuse we need to supply GCP credentials from the current user
      enabled = true
      runOptions = '--privileged -e GOOGLE_APPLICATION_CREDENTIALS=/home/root/.config/gcloud/application_default_credentials.json -v ${HOME}/.config/gcloud:/home/root/.config/gcloud'
    }
```
For the Google Batch profile solution I had made this configuration:
```
    docker {
      enabled = true
      runOptions = '--security-opt apparmor:unconfined --cap-add mknod --cap-add sys_admin --device=/dev/fuse'
    }
```

This is supposed to work as suggested by multiple posts regarding Docker privileged execution. However the program output suggests otherwise. I have checked the permissions of Compute Engine service account, it is set to Editor, my account as well, for that storage, if it counts.

### Program output 

![image](https://github.com/user-attachments/assets/8424973c-7627-42e5-9b5c-86aee99bf122)

### Environment 

* Nextflow version: v24.04.4

**Comment by pditommaso** [comment]
I don't think it's possible `gcsfuse` mount should be mounted via Google Batch API

**Reply by VasLem** [reply]
Thank you, when you say it is not possible, you mean it cannot be done through Nextflow, or Google batch just does not support this?

**Reply by pditommaso** [reply, marked as answer]
it cannot be done through Nextflow

**Reply by pditommaso** [reply]
You may want to have a look at [Fusion](https://www.nextflow.io/docs/latest/fusion.html#google-cloud). It should give access to any bucket, at task level, using the file path `/fusion/gs/your-bucket-name/your/data/path`
